### ISSUE TITLE
Swiper breaks on iOS10 safari with s.params is not an object (onTouchStart)

### ISSUE THREAD

**Issue by josh-sachs** [nolimits4web/Swiper#1979]
This is a (multiple allowed):
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* Swiper Version: EXACT RELEASE VERSION OR COMMIT HASH, HERE.
3.4.1
* Platform/Target and Browser Versions: PLATFORM CLIENT YOU ARE TARGETING SUCH AS macOS, Windows, CORDOVA, IOS, ANDROID, CHROME, ETC.
IOS10 Safari

### What you did
EXPLAIN WHAT YOU DID, PREFERABLY WITH CODE EXAMPLES, HERE.

TouchStart event shits the bed with (s.params is not an object)

### Expected Behavior
EXPLAIN WHAT IS TO BE EXPECTED, HERE.

Not to shit the bed.

### Actual Behavior
EXPLAIN WHAT IS ACTUALLY HAPPENING, HERE.

swiper.js line 1533 defines  
`s.onTouchStart = function (e) { ... }`
the first touch event on iOS safari, s is null, so line 1538 throws an error. 
`if (s.params.noSwiping && findElementInEvent(e, '.' + s.params.noSwipingClass)) { ... }`

if I modify the onTouchStart handler as such, it doesn't break - but not sure what the impact is. 
`s.onTouchStart = function (e) { 
if (!s) return;
... }`

The touchStart event appears to fire twice (at least on the first touch)  first time s === null second time s === swiper instance.

P.S. Remember, an issue is not the place to ask questions. You can use [Stack Overflow](http://stackoverflow.com/questions/tagged/swiper)
for that.

Before you open an issue, please check if a similar issue already exists or has been closed before.

**Comment by sconix** [comment]
This happens also in Chrome when simulating mobile and the same if check seems to fix it and there does not seem to be any drawbacks. Would be nice to get this fixed since its such a small fix.

**Comment by nolimits4web** [comment]
How to replicate the issue as i can't see such bug?

**Comment by sconix** [comment]
I can make few tests, but it might be that it requires more than one Swiper. I will test today that does it happen with one, but at least with two any swipe gesture will produce that error in Chrome with touch events enabled.

**Comment by sconix** [comment]
Sorry for the delay but dad a food poisoning so did not feel like doing anything for couple of days. So I did a quick test and it seems that it stats to happen only after call to swiper.destroy and by creating swiper again. So I guess the destroy process leaves something behind?

**Comment by snortblt** [comment]
FWIW, I'm seeing the same issue on iOS with a single swiper in an Ionic project right off the bat. No destroy being called (in my own code, anyway). Note this is a [ngx-swiper-wrapper](https://github.com/zefoy/ngx-swiper-wrapper) not Ionic's Slides wrapper.
`<swiper [config]="hSwiperOptions">
	<div class="swiper-slide swiper-slide-h" *ngFor="let s of [1,2,3,4,5,6,7]">					
              slide {{s}}
	</div>
</swiper>`

**Comment by sconix** [comment]
The ngx-swiper-wrapper is calling the destroy and thats why the bug exists when using that library. @nolimits4web would you like me to make new issue or even look into making PR for this? So the bug is definitely in the destroy not removing some event listener.

**Comment by sconix** [comment]
Finally had time to look into this and at least in my case this was due to bug in zone.js and not in Swiper. So at least I can't anymore reproduce this error after the latest version of zone.js.

**Comment by nolimits4web** [comment]
Issue is closed because of outdated/irrelevant/not actual/needed more information/inactivity.

If this issue is still actual and reproducible for latest version of Swiper, please create new issue and fill the issue template correctly:
* Clearly describe the issue including steps to reproduce when it is a bug.
* Make sure you fill in the earliest version that you know has the issue.
* Provide live link or JSFiddle/Codepen or website with issue
